test/parallel/test-net-allow-half-open.js

```diff
@@ -4,22 +4,44 @@ const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
-const server = net.createServer(common.mustCall((socket) => {
-  socket.end(Buffer.alloc(1024));
-})).listen(0, common.mustCall(() => {
-  const socket = net.connect(server.address().port);
-  assert.strictEqual(socket.allowHalfOpen, false);
-  socket.resume();
-  socket.on('end', common.mustCall(() => {
-    process.nextTick(() => {
-      // Ensure socket is not destroyed straight away
-      // without proper shutdown.
+{
+  const server = net.createServer(common.mustCall((socket) => {
+    socket.end(Buffer.alloc(1024));
+  })).listen(0, common.mustCall(() => {
+    const socket = net.connect(server.address().port);
+    assert.strictEqual(socket.allowHalfOpen, false);
+    socket.resume();
+    socket.on('end', common.mustCall(() => {
+      process.nextTick(() => {
+        // Ensure socket is not destroyed straight away
+        // without proper shutdown.
+        assert(!socket.destroyed);
+        server.close();
+      });
+    }));
+    socket.on('finish', common.mustCall(() => {
       assert(!socket.destroyed);
-      server.close();
-    });
+    }));
+    socket.on('close', common.mustCall());
   }));
-  socket.on('finish', common.mustCall(() => {
-    assert(!socket.destroyed);
+}
+
+{
+  const server = net.createServer(common.mustCall((socket) => {
+    socket.end(Buffer.alloc(1024));
```

Comment:
Is this needed now that the `end()` is also called on the client? Shouldn't this socket be resumed to consume the data sent by the client?

Reply:
Not sure, it does trigger the bug in the current form though...